emsdk

```diff
@@ -821,7 +834,8 @@ def build_fastcomp_tool(tool):
 
   enable_assertions = ENABLE_LLVM_ASSERTIONS.lower() == 'on' or (ENABLE_LLVM_ASSERTIONS == 'auto' and build_type.lower() != 'release' and build_type.lower() != 'minsizerel')
 
-  args = ['-DLLVM_TARGETS_TO_BUILD=X86;JSBackend', '-DLLVM_INCLUDE_EXAMPLES=OFF', '-DCLANG_INCLUDE_EXAMPLES=OFF', '-DLLVM_INCLUDE_TESTS=' + tests_arg, '-DCLANG_INCLUDE_TESTS=' + tests_arg, '-DLLVM_ENABLE_ASSERTIONS=' + ('ON' if enable_assertions else 'OFF')]
+  js_target = ';JSBackend' if build_js else ''
```

Comment:
How about this:

```python
only_supports_wasm = hasattr(tool, 'only_supports_wasm')

targets_to_build = 'X86'
if not only_supports_wasm: targets_to_build += ';JSBackend'
```

and then below:

```c++
if ENABLE_WASM or only_supports_wasm:
  args += ['-DLLVM_EXPERIMENTAL_TARGETS_TO_BUILD=WebAssembly']
```

That would then allow deleting the function `build_upstream_llvm_tool()` altogether (`ENABLE_WASM` variable can be left in whatever user-specified state in that case), and `def build_llvm_tool(tool, build_js):` -> `def build_llvm_tool(tool):`

Reply:
Sure, done